lib/global-styles.php

```diff
@@ -76,60 +76,60 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
  * @return array New block editor settings
  */
 function gutenberg_experimental_global_styles_settings( $settings ) {
-	$origin = 'theme';
+	// Set what is the context for this data request.
+	$context = 'all';
 	if (
+		is_callable( 'get_current_screen' ) &&
+		function_exists( 'gutenberg_is_edit_site_page' ) &&
+		gutenberg_is_edit_site_page( get_current_screen()->id ) &&
 		WP_Theme_JSON_Resolver::theme_has_support() &&
 		gutenberg_supports_block_templates()
 	) {
-		// Only lookup for the user data if we need it.
-		$origin = 'user';
+		$context = 'site-editor';
 	}
-	$tree = WP_Theme_JSON_Resolver::get_merged_data( $settings, $origin );
-
-	// STEP 1: ADD FEATURES
-	//
-	// These need to be always added to the editor settings,
-	// even for themes that don't support theme.json.
-	// An example of this is that the presets are configured
-	// from the theme support data.
-	$settings['__experimentalFeatures'] = $tree->get_settings();
-
-	// STEP 2 - IF EDIT-SITE, ADD DATA REQUIRED FOR GLOBAL STYLES SIDEBAR
-	//
-	// In the site editor, the user can change styles, so the client
-	// needs the ability to create them. Hence, we pass it some data
-	// for this: base styles (core+theme) and the ID of the user CPT.
+
+	if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
+		$context = 'rest-request';
+	}
+
+	$origin = 'theme';
 	if (
-		is_callable( 'get_current_screen' ) &&
-		function_exists( 'gutenberg_is_edit_site_page' ) &&
-		gutenberg_is_edit_site_page( get_current_screen()->id ) &&
 		WP_Theme_JSON_Resolver::theme_has_support() &&
 		gutenberg_supports_block_templates()
 	) {
+		// Only lookup for the user data if we need it.
+		$origin = 'user';
+	}
+	$consolidated = WP_Theme_JSON_Resolver::get_merged_data( $settings, $origin );
+
+	if ( 'rest-request' === $context ) {
+		$settings['__experimentalStyles'] = $consolidated->get_raw_data()['styles'];
```

Comment:
Could this be:
```suggestion
		$settings['__experimentalGlobalStylesBaseStyles'] = $consolidated->get_raw_data();
```

To get the whole consolidated raw data and use the same name.

Reply:
Was thinking of this:

- consolidated settings should be under `__experimentalFeatures` (which already are)
- consolidated styles should be under `__experimentalStyles` (this PR)
- are there any other fields meaningful for mobile (templateParts, etc)?

We also need to update `__experimentalGlobalStylesBaseStyles`: needs a better name (it passes down styles + settings) and filtering (some data shouldn't be there).  This is also not the consolidated data, but only the data for the theme, which I understand is not what mobile needs?

Reply:
After double-checking, we are good with having `__experimentalStyles` and the consolidated settings under `__experimentalFeatures` 👍